### PR TITLE
Finish automating relases through CI

### LINKED ISSUE #388: [FEATURE] Configure automated weekly releases
As discussed in our strategy meeting today, we plan to cut weekly releases of the ongoing development on `unstable`, which we can then promote to major/minor/patch releases whenever we think its justified.

I'll update here with details of the planned release process before I begin implementation.

### LINKED ISSUE #442: Automate release packaging and uploads through github actions
Towards #388 

Uploading releases is currently automated, but we'd like for our CI to take care of this for us, rather than having to run the script manually and deal with the vagaries of each operators environment.

We want a githug workflow that will be triggered by [version tags](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags) that will act on release candidate commits to package the release and upload the artifact.

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #388 
Closes #442

This PR adds the finishing touches needed to push the release automation through.  It also expands the documentation, in the hopes that I've provided enough pointers that anyone can pickup maintenance and manage the releases without needing to retrace my steps.

Please especially review [the documentation on releasing](https://github.com/informalsystems/apalache/blob/3f0e0bd8779cc3d862ec4ecb88328c81c301364a/CONTRIBUTING.md#releases), to be sure it all seems clear.

I've tested the pipeline on my fork. See 

- The automatically created release PR (release notes suppressed to reduce noise during testing): https://github.com/shonfeder/apalache/pull/11
- And the release generated on merging the PR: https://github.com/shonfeder/apalache/releases/tag/v0.0.2-test

- [ ] ~Tests added for any new code~
- [X] Documentation added for any new functionality
- [ ] ~Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality~

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -249,14 +249,18 @@ The process proceeds in two steps:
...
    notes. These are triggered every Monday by a cron job or [manually via the
    GitHub UI][github-ui].
-2. A human reviews the PR, approves it, and merges into the trunk, at which
-   point CI kicks in to:
+   - The scheduled releases increment the patch number.
+   - Use the _Version_ input field to manually specify the version to release.
+
+   <img src="./trigger-release.png" alt="How to trigger a release" width="700px">
+2. A human reviews the PR, approves it, and **merges** (_DO NOT SQUASH OR
+   REBASE_) into the trunk, at which point CI kicks in to:
```
Comment (konnov) [inline CONTRIBUTING.md:257]: Can we disable squash merge? Just asking :-)

File: script/release-prepare.sh [hunk 2]
```diff
@@ -3,13 +3,24 @@ set -euo pipefail
 
 set -o xtrace
 
-# Perpare a release on the current branch
+# Perpare a release on the current branch.
```
Comment (konnov) [inline script/release-prepare.sh:6]:
```suggestion
# Prepare a release on the current branch.
```

File: script/release-prepare.sh [hunk 3]
```diff
@@ -29,7 +41,10 @@ else
     RELEASE_VERSION=$("$DIR"/get-version.sh)
 fi
 
+# Prepare the release on a new branch
 git checkout -b "release/${RELEASE_VERSION}"
+
+# Generatre the release notes
```
Comment (konnov) [inline script/release-prepare.sh:47]:
```suggestion
# Generate the release notes
```

File: script/release-publish.sh [hunk 4]
```diff
@@ -4,22 +4,25 @@ set -euo pipefail
 set -o xtrace
 
 # Package a release and tag the commit
+#
+# NOTE: While this script can be run locally, is mainly desgined for use in our
```
Comment (konnov) [inline script/release-publish.sh:8]:
```suggestion
# NOTE: While this script can be run locally, is mainly designed for use in our
```

File: script/update-changes.sh [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 # Run this script to update the CHANGES.md with the RELEASE-NOTES.md used in
 # the last release.
+#
+# NOTE: While this script can be run locally, is mainly desgined for use in our
```
Comment (konnov) [inline script/update-changes.sh:6]:
```suggestion
# NOTE: While this script can be run locally, is mainly designed for use in our
```

**Review by konnov — APPROVED** [review]
LGTM. Just a few spelling errors.

File: CONTRIBUTING.md (hunk 1 above)
Comment (shonfeder) [reply CONTRIBUTING.md:257]: Sure: done!

I'll still leave this warning in place, in case the policies change in the future.

**Comment by shonfeder** [comment]
Sorry for the typos and thanks for the fixes!

**Comment by shonfeder** [comment]
Docker test failed due to flaky availability of dependency:

```
Error:  Failed to execute goal on project tla-import: Could not resolve dependencies for project at.forsyte.apalache:tla-import:jar:0.8.2-SNAPSHOT: Could not find artifact org.lamport:tla2tools:jar:1.7.0-SNAPSHOT in oss-sonatype (https://oss.sonatype.org/content/repositories/snapshots/) -> [Help 1]
```

None of this code effects the dockerized build, and it already passed before a trivial change, so I'm gonna merge anyhow once the other tests pass.

**Pull request merged**